probe-rs/src/architecture/arm/core/m4.rs

```diff
@@ -635,6 +635,33 @@ impl<'probe> CoreInterface for M4<'probe> {
     fn architecture(&self) -> Architecture {
         Architecture::Arm
     }
+
+    /// Read the hardware breakpoints from FpComp registers, and adds them to the Result Vector. A value of 0 in any position of the Vector indicates that the position is unset/available.
```

Comment:
Please use an `Option<u32>` and `None` for missing entries and `Some(n)` for present entries.
This does not hinder perfromance, but improves readablity a lot.

Reply:
@Yatekii ... I've pushed updates to other feedback, but would appreciate a bit of 'negotiation' on this one. The FpComp registers can have broken sequences of actual breakpoints and zeros. As I iterate over the retrieved values, I personally think it is at least as clean to look for the first available next breakpoint as `if breakpoint.is_zero()` 

Or was there a different case of readability that I'm missing?

Reply:
If you use `0` as not present, you can mistakenly read that value as a valid address loaction. With an `Option` you cannot do that. In my optinion that's the biggest value Rust provides over other languages.
`0` is really not a good value to encode things if it is otherwise possible :)

Reply:
Gotcha ... I will consider myself 'negotiated' :)

Reply:
Glad we agreed on this =D

Reply:
@Yatekii, this is not implemented.